Update OptionCard component

eldiegod opened the pull request:
fixes: #1676

figma: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=4118%3A144975

eldiegod committed:
Update OptionCard component

fixes: #1676

figma: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=4118%3A144975

rafalpawlow reviewed (changes requested):
In figma, radio component doesn't inherit error state
![Zrzut ekranu 2022-01-13 o 18 49 39](https://user-images.githubusercontent.com/13706246/149382755-55cb7247-999f-422d-8301-a2f4e12dd459.png)

Another thing is, in figma we have two types of option card `Radio` and `Checkbox`. Second one is missing in your PR.

rafalpawlow commented on src/components/_inputs/OptionCard/OptionCard.stories.tsx:
```diff
@@ -21,9 +23,10 @@ const Template: Story<OptionCardProps> = (args) => {
   return (
-    <div style={{ display: 'grid', gridTemplateColumns: 'auto auto', gap: '16px' }}>
-      <OptionCard {...args} value="1" selectedValue={selected} onClick={handleClick} />
+    <div style={{ display: 'grid', gridTemplateColumns: 'auto auto auto', gap: '16px' }}>
```
In figma we have 2 gaps depending on the breakpoint `16px` and `24px`. Also I think we should move `Option card group` as a separate component.

rafalpawlow commented:
Figma says: `In the Option card component, the display limit for both the main label is set to 1 line of text. Upon exceeding this limit, the text should be trimmed and ended with an ellipsis.`.

But current implementation allows 2 lines of text
![Zrzut ekranu 2022-01-13 o 19 00 47](https://user-images.githubusercontent.com/13706246/149384390-91608566-06f6-4797-baf6-8953faf9c9eb.png)

rafalpawlow commented:
I'm not sure about spacing between text and hovered input. @toiletgranny is that fine?
![Zrzut ekranu 2022-01-13 o 19 09 48](https://user-images.githubusercontent.com/13706246/149385456-b4e3b0de-7cb5-484f-8da6-070cd2cfb584.png)

rafalpawlow commented:
Border colors are wrong
we should use the following:

**white**
default: `colorBorderAlpha`
hover: `colorBorderStrongAlpha`
pressed: `colorBorderMutedAlpha`

**error**
hover: `colorCoreRed300`
default: `colorCoreRed500`
pressed: `colorCoreRed700`

currently we use `colorCoreRed400` for all variants

**blue**
hover: `colorCoreBlue300`
pressed: `colorCoreBlue700`

toiletgranny commented:
I agree with everything @rafalpawlow pointed out here, and to answer/comment on some of the points he brought up:
1. Yes, the spacing between the label and the Radio button/Checkbox should be `16px`, awesome catch! This can be inspected in Figma, though it might be tricky. 
![CleanShot 2022-01-14 at 11 02 41](https://user-images.githubusercontent.com/35307309/149497302-5b9d954f-f9e2-4b19-9f39-5647d82b0c5f.png)

2. It's great to see text & icon colors using our new design tokens, but @rafalpawlow is right that we forgot about borders. Especially in "the error state". In case that's helpful, @eldiegod I put up this quick prototype for you as a quick reference on how border colors are changing upon hovering and pressing the Option card component.


https://www.figma.com/proto/Pf31tuYpozYmpq163U2ho8/Web-Components?page-id=4645%3A137109&node-id=4645%3A137110&viewport=396%2C48%2C1&scaling=min-zoom&starting-point-node-id=4645%3A137110

eldiegod committed:
cr

eldiegod commented:
@rafalpawlow @toiletgranny I think I have addressed all the comments except for the optionCardGroup components as I'm not sure it should be done here or it's a matter for another ticket.

Saying it because the initial complexity scope for this ticket was a 2 but it has gotten a bit intricate so not sure if we just missed that or we are doing more than planned

rafalpawlow commented:
@eldiegod component still allows for 2 lines of title

eldiegod committed:
fix lina clamp

eldiegod committed:
ignore line

rafalpawlow commented:
Something is wrong with input when there is a lot of text ant card is in `active` state
![Zrzut ekranu 2022-01-17 o 15 53 11](https://user-images.githubusercontent.com/13706246/149791781-446f3ea6-d65f-4dda-ba99-6d175a31fad9.png)

eldiegod commented:
> Something is wrong with input when there is a lot of text ant card is in `active` state

Can you tell me exactly what you mean with wrong? @rafalpawlow

eldiegod committed:
layout fix

rafalpawlow commented:
@eldiegod Look at the screenshot, radio button is not round, it is egg-shaped

eldiegod commented:
> @eldiegod Look at the screenshot, radio button is not round, it is egg-shaped

wanted to make sure, I think I fixed it now @rafalpawlow

rafalpawlow reviewed (approved):
Looks good! However I found one more thing which we probably should tackle in another issue. Pressed radio component looks a little bit different in figma:

![Zrzut ekranu 2022-01-18 o 13 04 19](https://user-images.githubusercontent.com/13706246/149934320-a88d31cc-28ec-474f-baf3-f46c218a6545.png)
![Zrzut ekranu 2022-01-18 o 13 04 57](https://user-images.githubusercontent.com/13706246/149934322-597c744f-5de2-419a-826d-95f2f0d391b0.png)

eldiegod commented:
> Looks good! However I found one more thing which we probably should tackle in another issue. Pressed radio component looks a little bit different in figma:
> 
> ![Zrzut ekranu 2022-01-18 o 13 04 19](https://user-images.githubusercontent.com/13706246/149934320-a88d31cc-28ec-474f-baf3-f46c218a6545.png) ![Zrzut ekranu 2022-01-18 o 13 04 57](https://user-images.githubusercontent.com/13706246/149934322-597c744f-5de2-419a-826d-95f2f0d391b0.png)

@rafalpawlow Awesome! Yep that's right, that would be more of an existing issue with the `RadioInput` component itself

kdembler commented:
@eldiegod Could you please create an issue for this difference? We may have missed that
